test/mercury/camelcase-template-parameters.test.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable no-undef */
+"use strict";
+import { validateFile } from "../../src/validator";
+import {
+  getHappySpec,
+  conforms,
+  breaksOnlyOneRule,
+  breaksTheseRules,
+  renameKey,
+  renderSpecAsFile
+} from "../utils.test";
+
+const PROFILE = "mercury";
+const TEMPLATE_RULE =
+  "http://a.ml/vocabularies/data#camelcase-template-parameters";
+
+describe("template parameter checking tests", () => {
+  it("fails when template starts with capital", async () => {
+    const doc = getHappySpec();
+    renameKey(doc["/resource"], "/{resourceId}", "/{ResourceId}");
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+    breaksOnlyOneRule(result, TEMPLATE_RULE);
+  });
+
+  it("fails when template starts with number", async () => {
+    const doc = getHappySpec();
+    renameKey(doc["/resource"], "/{resourceId}", "/{9esourceId}");
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+    breaksOnlyOneRule(result, TEMPLATE_RULE);
+  });
+
+  it("fails when template has a space", async () => {
+    const doc = getHappySpec();
+    renameKey(doc["/resource"], "/{resourceId}", "/{resource Id}");
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+    breaksOnlyOneRule(result, TEMPLATE_RULE);
+  });
+
+  it("fails when template has a space", async () => {
```

Comment:
Looks like a duplicate of the test above

Reply:
Also, I was curious why call `getHappySpec()` in each test instead of `beforeEach`

Reply:
Dude, it's an important test! Test twice, cut once, right? Test twice, code once? I dunno. Removed.